src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -0,0 +1,231 @@
+          role="dialog"
+          __css={styles.menuContainer}
+          {...(!isOpen && { display: "none" })}
+        >
+          <UnorderedList
+            styleType="none"
+            marginInlineStart="0"
+            __css={styles.menu}
+          >
+            {isOpen &&
+              items.map((item: MultiSelectItem) => (
+                <ListItem key={item.id} py={1}>
+                  {item.children ? (
+                    <>
+                      <Checkbox
+                        id={item.id}
+                        labelText={item.name}
+                        showLabel={true}
+                        name={item.name}
+                        // If onMixedStateChange is not passed as a prop, handle
+                        // the parent checkbox as a regular checkbox using onChange
+                        {...(onMixedStateChange !== undefined
+                          ? {
+                              isChecked: isAllChecked(id, item) || false,
+                              isIndeterminate:
+                                isIndeterminate(id, item) || false,
+                              onChange: onMixedStateChange,
+                            }
+                          : {
+                              isChecked: isChecked(id, item.id) || false,
+                              onChange: onChange,
+                            })}
+                      />
+                      <UnorderedList
+                        styleType="none"
+                        marginInlineStart="0"
+                        __css={styles.menuChildren}
+                      >
+                        {item.children.map((childItem) => {
+                          return (
+                            <ListItem key={childItem.id} py={1}>
+                              <Checkbox
+                                id={childItem.id}
+                                labelText={childItem.name}
+                                showLabel={true}
+                                name={childItem.name}
+                                isChecked={isChecked(id, childItem.id) || false}
+                                onChange={onChange}
+                              />
+                            </ListItem>
+                          );
+                        })}
+                      </UnorderedList>
+                    </>
+                  ) : (
+                    <Checkbox
+                      id={item.id}
+                      labelText={item.name}
+                      showLabel={true}
+                      name={item.name}
+                      isChecked={isChecked(id, item.id) || false}
+                      onChange={onChange}
+                    />
+                  )}
+                </ListItem>
+              ))}
+          </UnorderedList>
+          {isOpen && !isMobile && (
+            <Stack direction="row" spacing={4} justify="flex-end">
+              <Button
+                id="multiselect-dialog-clear"
```

Comment:
I think I've seen examples where there are multiple `MultiSelect` components on a page. If this id is static, then this will cause an accessibility issue because there will be multiple buttons with the same id value. This should be updated to use the `id` prop value passed. This comment applies to other `id` props that have a static value.

Reply:
fixed

Reply:
fixed